docs/dev/Pagination.md

```diff
@@ -0,0 +1,27 @@
+# Opendistro ODBC Driver Cursor (Pagination) Support Design Documentation
+
+## Overview
+Elasticsearch ODBC Driver supports forward-only cursor. This document illustrates how the cursor(pagination) is handled in the driver. 
+
+For information on how the pagination is supported on Elasticsearch server, check [Opendistro SQL Cursor (Pagination) Support](https://github.com/opendistro-for-elasticsearch/sql/blob/master/docs/dev/Pagination.md).
+
+## Data Flow
+<p align="center"> 
+<img src="img/data_flow.png" width="60%">
+</p>
+
+* Steps 3 & 4 will repeatedly get datarows until entire result is retrieved.
+* Step 5 will send a request to close cursor whenever the connection is closed.
+* ODBC Driver will provide an option to define fetch size as a connection parameter.
+* If fetch size is not given or zero then the number of rows per request will be as per server-defined [default fetch size](https://github.com/opendistro-for-elasticsearch/sql/blob/master/docs/dev/Pagination.md#42-salient-points).
```

Comment:
If the `fetch_size` is explicitly zero, we will default back to non-cursor behavior,  which means there will be only one page with no `cursor` field.

Reply:
When fetch_size is zero/null, the request will not include fetch_size. So fetch_size will never be explicitly zero. I will rewrite the sentence to make it more clear.

Reply:
If user provides a value of `zero` as `fetch_size`, we should include `fetch_size` as zero in the request. In case no value is passed, `fetch_size` would be null, and thus need not be included in the request.

Reply:
Will add an explicit zero case